client/my-sites/domains/domain-management/components/primary-domain-selector/index.tsx

```diff
@@ -0,0 +1,184 @@
+				return domain.is_wpcom_staging_domain;
+			}
+
+			return true;
+		} );
+	}
+
+	const onSelectChange = ( event: ChangeEvent< HTMLSelectElement > ) => {
+		setSelectedDomain( event.target.value );
+	};
+
+	const onSubmit = ( event: FormEvent< HTMLFormElement > ) => {
+		event.preventDefault();
+
+		const domain = domains.find( ( d ) => d.domain === selectedDomain );
+		if ( ! domain || ! selectedDomain ) {
+			return;
+		}
+
+		setIsSettingPrimaryDomain( true );
+		onSetPrimaryDomain( selectedDomain, () => setIsSettingPrimaryDomain( false ), domain.type );
+	};
+
+	const trackUpgradeClick = () => {
+		recordTracksEvent( 'calypso_primary_site_address_nudge_cta_click', {
+			cta_name: 'add_custom_domain',
+		} );
+	};
+
+	return (
+		<FormFieldset className="domains-set-primary-address">
+			<FormLabel htmlFor="primary-domain-selector" className="domains-set-primary-address__title">
+				{ translate( 'Primary site address' ) }
+			</FormLabel>
+			<div>
+				<FormSettingExplanation className="domains-set-primary-address__subtitle">
+					{ translate(
+						'The current primary address set for this site is: {{strong}}%(selectedDomain)s{{/strong}}.',
+						{
+							args: { selectedDomain: primaryDomain as string },
+							components: {
+								strong: <strong />,
+							},
+						}
+					) }{ ' ' }
+					{ validPrimaryDomains.length > 1
+						? translate(
+								'You can change it by selecting a different address from the list below. {{learnMoreLink}}Learn more{{/learnMoreLink}}.',
+								{
+									components: {
+										learnMoreLink: (
+											<InlineSupportLink supportContext="primary-site-address" showIcon={ false } />
+										),
+									},
+								}
+						  )
+						: translate(
+								'Before you can change the primary site address you have to add a new custom domain. Buy a {{domainSearchLink}}custom domain{{/domainSearchLink}} or {{mapDomainLink}}map{{/mapDomainLink}} a domain you already own.',
+								{
+									components: {
+										domainSearchLink: (
+											<a
+												href={ '/domains/add/use-my-domain/' + primaryDomain }
+												onClick={ trackUpgradeClick }
+											/>
+										),
+										mapDomainLink: (
+											<a href={ '/domains/add/' + primaryDomain } onClick={ trackUpgradeClick } />
+										),
+									},
+								}
+						  ) }
+				</FormSettingExplanation>
+				{ validPrimaryDomains.length > 1 && (
+					<>
+						<FormSelect
+							disabled={ isSettingPrimaryDomain }
+							id="primary-domain-selector"
+							onChange={ onSelectChange }
+							value={ selectedDomain }
+						>
+							{ validPrimaryDomains.map( ( domain ) => (
+								<option key={ domain.domain } value={ domain.domain }>
+									{ domain.domain }
+								</option>
+							) ) }
+						</FormSelect>
+
+						<FormButton
+							className="domains-set-primary-address__submit"
+							primary
+							busy={ isSettingPrimaryDomain }
+							disabled={ isSettingPrimaryDomain }
```

Comment:
Presumably should also be disabled when the current primary domain is the same as the selected domain?

Reply:
The current primary domain is not showed in the dropdown.